### PR TITLE
Log every user action in the catalog

### PR DESCRIPTION
**Pull request by fviale**
Also:
  - ensure that authentication and RBAC is enforced in delete empty buckets
  - fix tests in fast machines (add tiny waits to avoid 2 revisions with the same commit time in ms)

### CODE REVIEW

File: src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java [hunk 1]
```diff
@@ -424,6 +436,9 @@ public void testUpdateObjectMetadataOnlyKindOrOnlyContentTypeOrOnlyProjectNameOr
         assertThat(catalogObjectUpdatedProjectName.getMetadataList()).hasSize(2);
         assertThat(catalogObjectUpdatedProjectName.getCommitTimeRaw()).isNotEqualTo(String.valueOf(now));
 
+        // a necessary sleep to avoid that the new revision share the same commit time (and thus is discarded)
+        Thread.sleep(2);
```
Comment (mbenguig) [inline src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java:440]: tinyWait is not accessible from there ?

**Review by mbenguig — APPROVED** [review]
A very  "tiny" comment.

File: src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java (hunk 1 above)
Comment (fviale) [reply src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java:440]: I can create a method yes

**Comment by fviale** [comment]
Retest this please

**Review by AlexFalessi — APPROVED** [review]

**Pull request merged**